cmd/kube-apiserver/app/server.go

```diff
@@ -299,7 +299,15 @@ func Run(s *options.ServerRunOptions) error {
 	}
 
 	admissionControlPluginNames := strings.Split(s.GenericServerRunOptions.AdmissionControl, ",")
-	pluginInitializer := kubeadmission.NewPluginInitializer(client, sharedInformers, apiAuthorizer)
+	var cloudConfig []byte
+
+	if s.CloudProvider.CloudConfigFile != "" {
+		cloudConfig, err = ioutil.ReadFile(s.CloudProvider.CloudConfigFile)
```

Comment:
Okay passing the file path is not enough to centralize the initialization of cloudprovider in one place (namely `SetCloudConfig`), we would need name of provider too, so we would have to pass 2 params? :( The problem with passing 2 additional params to admission plugin initialize is - now it will have 5 parameters. Nothing too bad but not clean either.

The initialization of cloudprovider in admission plugin is bit adhoc - https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/persistentvolume/label/admission.go#L74 and I don't like it but I am thinking a proper fix is to simply pass initialized `cloudprovider.Interface` object directly into the plugin rather than pass ConfigPath etc. 

So I tried doing that and it has some spill over effect - like we don't always have cloudprovider initialized even in api server and I need to refactor apiserver some more to make that happen. 

At this late in 1.6 cycle, I am bit hesitant to make bigger changes and I can confirm with some certainity that fix indeed works and I was able to create PVs in multizones after this change.

@saad-ali @msau42 I think the code here was already bit non-ideal. I would like to talk to @jsafrane and understand why he didn't pass cloudprovider object directly and used ad-hoc cloudprovider initialization.

In the meanwhile, I think we should merge this - so as underlying problem is fixed. I can work on cleaning it up.